File: FlinkScalaCode/src/main/scala/com/wubaibao/flinkscala/code/chapter6/partitions/GlobalTest.scala
package com.wubaibao.flinkscala.code.chapter6.partitions

import org.apache.flink.streaming.api.scala.{DataStream, StreamExecutionEnvironment}

/**
 * Flink Global全局分区测试
 */
object GlobalTest {
  def main(args: Array[String]): Unit = {
    val env: StreamExecutionEnvironment = StreamExecutionEnvironment.getExecutionEnvironment
    // 导入隐式转换
    import org.apache.flink.api.scala._
    val ds: DataStream[Int] = env.fromCollection(List(1, 2, 3, 4, 5, 6, 7, 8, 9, 10))
    ds.print("普通打印")
    // 设置全局分区
    ds.global.print("global")
    env.execute()
  }

}

File: FlinkJavaCode/src/main/java/com/wubaibao/flinkjava/code/chapter6/partitions/GlobalTest.java
package com.wubaibao.flinkjava.code.chapter6.partitions;

import org.apache.flink.configuration.RestOptions;
import org.apache.flink.streaming.api.datastream.DataStreamSource;
import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;

import java.util.Arrays;

/**
 * Flink global Partition 测试
 */
public class GlobalTest {
    public static void main(String[] args) throws Exception {
        StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
        DataStreamSource<Integer> ds = env.fromCollection(Arrays.asList(1, 2, 3, 4, 5, 6, 7, 8, 9, 10));
        ds.global().print("global");
        env.execute();
    }
}
